[TLVB-159] 카드 클릭 비동작 현상 수정, 카드 클릭 시 상세 이벤트 조회 페이지 이동 구현, 좋아요, 즐겨찾기 이벤트 API 연동

JengYoung opened the pull request:
## PR 설명

- 현재 상세 이벤트 조회 페이지 세부 로직 구현 중이며, 카드 클릭 시 이벤트가 정상 작동되지 않는 현상을 해결하였습니다.
- 원인은 `React.createPortal`과 `fixed position`의 특성이 합쳐져 생긴 이슈였습니다.

## 기타 수정 사항

- `MainContainer`의 `main` 태그가 2개가 모달로 인해 발생하였습니다.
- 따라서 `main` 태그를 `div`로 바꿀 수 있도록 `as prop`으로 동적으로 태그를 제어하는 방식으로 해결하였습니다.
- `id` selector 역시 `class`로 바꾸었습니다.

## 체크 리스트

- [x] 사용자는 비로그인을 하여도 전체 이벤트를 볼 수 있다.
- [x] 사용자는 로그인을 하여도 전체 이벤트를 볼 수 있다.
- [x] 메인 페이지에서 카드를 클릭하면 상세 이벤트 보기 페이지로 이동할 수 있다.
- [x] 사용자는 참여 버튼을 누르면 이벤트에 참여할 수 있다.
- [x] 사용자는 참여 완료 시 리뷰하기 버튼이 생긴다.
- [x] 사용자는 리뷰하기 버튼을 클릭 시 리뷰를 할 수 있다.
- [x] 사용자는 좋아요를 누를 시 이벤트를 좋아요로 추가할 수 있다.
- [x] 사용자는 좋아요 취소를 누를 시 이벤트의  좋아요를 취소할 수 있다.
- [x] 사용자는 즐겨찾기를 누를 시 이벤트의 가게를 즐겨찾기 할 수 있다.
- [x] 사용자는 즐겨찾기 취소를 누를 시 이벤트의 가게 즐겨찾기를 취소할 수 있다.
- [x] 사용자는 주소를 수정할 수 있다.
- [x] 사용자는 리뷰 리스트를 볼 수 있다.
- [x] 사용자는 리뷰를 클릭하면 리뷰 전체 내역을 볼 수 있다.
- [x] 리뷰가 없는 이벤트 상세 조회 페이지에는 기본적으로 리뷰를 장려하는 카드가 보인다.
 
## 참고 자료

- **해당 오류에 대한 이슈 트래킹 해결 과정을 다음 글로 남겼습니다.**
- https://oranjik.notion.site/You-Don-t-Know-createPortal-fixed-position-1062694027ba44829532984d1c0e60ab

## 시연 영상

- [x] 사용자는 비로그인을 하여도 전체 이벤트를 볼 수 있다.
- [x] 사용자는 로그인을 하여도 전체 이벤트를 볼 수 있다.
- [x] 메인 페이지에서 카드를 클릭하면 상세 이벤트 보기 페이지로 이동할 수 있다.

![상세 이벤트 조회](https://user-images.githubusercontent.com/78713176/146630468-8b887a8b-720a-4cef-807f-7599d3281149.gif)

- [x] 사용자는 참여 완료 시 리뷰하기 버튼이 생긴다.
- [x] 사용자는 리뷰하기 버튼을 클릭 시 리뷰를 할 수 있다.

![참여 완료 로직 구현](https://user-images.githubusercontent.com/78713176/146634585-d2053b92-53f0-44dc-a3da-99bceae14904.gif)

- [x] 사용자는 좋아요를 누를 시 이벤트를 좋아요로 추가할 수 있다.
- [x] 사용자는 좋아요 취소를 누를 시 이벤트의  좋아요를 취소할 수 있다.
- [x] 사용자는 즐겨찾기를 누를 시 이벤트의 가게를 즐겨찾기 할 수 있다.
- [x] 사용자는 즐겨찾기 취소를 누를 시 이벤트의 가게 즐겨찾기를 취소할 수 있다.

> ### 현재 API 이상 현상  
> 좋아요 및 즐겨찾기가 제대로 적용되지 않는 현상 발생. - 백엔드에서 내일까지 해결하기로 하였습니다.

![즐겨찾기 좋아요](https://user-images.githubusercontent.com/78713176/146641629-513484d0-d9a0-4054-ac7b-f035c838b836.gif)
![좋아요 즐겨찾기 구현2](https://user-images.githubusercontent.com/78713176/146641654-ec8ea2b8-908d-4afb-a93c-c7a1608372d9.gif)

- [x] 사용자는 주소를 수정할 수 있다.

![주소 수정](https://user-images.githubusercontent.com/78713176/146641687-0467e378-fd37-43f2-bcfe-c194829182f4.gif)

- [x] 사용자는 리뷰 리스트를 볼 수 있다.
- [x] 사용자는 리뷰를 클릭하면 리뷰 전체 내역을 볼 수 있다.
- [x] 리뷰가 없는 이벤트 상세 조회 페이지에는 기본적으로 리뷰를 장려하는 카드가 보인다.

![리뷰 보기](https://user-images.githubusercontent.com/78713176/146641779-df08d49f-eb03-4bf1-9a55-c498c42b6e39.gif)

JengYoung committed:
fix: 카드 클릭 비동작 현상 수정, 카드 클릭 시 상세 이벤트 조회 페이지 이동 구현

커밋 설명

- 현재 상세 이벤트 조회 페이지 세부 로직 구현 중이며, 카드 클릭 시 이벤트가 정상 작동되지 않는 현상을 해결하였습니다.
- 원인은 `React.createPortal`과 `fixed position`의 특성이 합쳐져 생긴 이슈였습니다.

기타 수정 사항

- `MainContainer`의 `main` 태그가 2개가 모달로 인해 발생하였습니다.
- 따라서 `main` 태그를 `div`로 바꿀 수 있도록 `as prop`으로 동적으로 태그를 제어하는 방식으로 해결하였습니다.
- `id` selector 역시 `class`로 바꾸었습니다.

체크 리스트

- [ ] 사용자는 비로그인을 하여도 전체 이벤트를 볼 수 있다.
- [ ] 사용자는 로그인을 하여도 전체 이벤트를 볼 수 있다.
- [x] 메인 페이지에서 카드를 클릭하면 상세 이벤트 보기 페이지로 이동할 수 있다.
- [ ] 사용자는 참여 버튼을 누르면 이벤트에 참여할 수 있다.
- [ ] 사용자는 참여 완료 시 리뷰하기 버튼이 생긴다.
- [ ] 사용자는 리뷰하기 버튼을 클릭 시 리뷰를 할 수 있다.

참고 자료

- 해당 오류에 대한 이슈 트래킹 해결 과정을 다음 글로 남겼습니다.
- https://oranjik.notion.site/You-Don-t-Know-createPortal-fixed-position-1062694027ba44829532984d1c0e60ab

JengYoung committed:
feat: 실제 API 연동을 통한 상세 이벤트 정보 조회 기능 구현

JengYoung committed:
feat: participateEvent 상태관리 및 api 생성

JengYoung committed:
feat: 이벤트 참여 기능 구현

- 커밋 설명
사용자는 이벤트 참여 버튼을 누를 시 이벤트를 참여할 수 있다.
참여 완료 시, 알림창이 뜬다.

- 체크 리스트
[x] 사용자는 비로그인을 하여도 전체 이벤트를 볼 수 있다.
[x] 사용자는 로그인을 하여도 전체 이벤트를 볼 수 있다.
[x] 메인 페이지에서 카드를 클릭하면 상세 이벤트 보기 페이지로 이동할 수 있다.
[x] 사용자는 참여 버튼을 누르면 이벤트에 참여할 수 있다.
[ ] 사용자는 참여 완료 시 리뷰하기 버튼이 생긴다.
[ ] 사용자는 리뷰하기 버튼을 클릭 시 리뷰를 할 수 있다.

JengYoung committed:
feat: 이벤트 참여 완료 API 및 전역 상태 로직 구현

JengYoung committed:
feat: 이벤트 상세 보기 페이지 로직 요구사항 모두 구현

- 커밋 설명
해당 체크 리스트에 대한 요구 사항을 모두 만족하는 로직을 구현하였다.

[x] 사용자는 비로그인을 하여도 전체 이벤트를 볼 수 있다.
[x] 사용자는 로그인을 하여도 전체 이벤트를 볼 수 있다.
[x] 메인 페이지에서 카드를 클릭하면 상세 이벤트 보기 페이지로 이동할 수 있다.
[x] 사용자는 참여 버튼을 누르면 이벤트에 참여할 수 있다.
[x] 사용자는 참여 완료 시 리뷰하기 버튼이 생긴다.
[x] 사용자는 리뷰하기 버튼을 클릭 시 리뷰를 할 수 있다.

- 참고 사항
현재 API가 제대로 완료되지 않았으므로, 세부 설정은 이후 API가 완성된 후 수정한다.

JengYoung committed:
fix: build 에러 수정, 기타 코드 리팩토링

JengYoung committed:
feat: 좋아요, 즐겨찾기 실제 API 연동 로직 구현

JengYoung committed:
fix: 좋아요, 즐겨찾기 API 미작동 현상 수정

JengYoung committed:
feat: 주소 수정 기능 구현

MiranLeeeee reviewed (approved):
Jason 구현사항 확인하였습니당!
집에 가서 더 자세히 보도록 하겠습니다,,
고생많으셨어요👍

JengYoung committed:
feat: 이벤트 상세 조회 페이지 리뷰 조회 기능 구현

JengYoung committed:
feat: 리뷰가 없을 시 기본적으로 보이는 리뷰 없음 카드 구현

- 구현 내용
리뷰가 없을 시 리뷰를 작성할 수 있도록 하는 카드를 구현한다.

- 현재까지의 요구사항 체크 리스트
- [x] 사용자는 비로그인을 하여도 전체 이벤트를 볼 수 있다.
- [x] 사용자는 로그인을 하여도 전체 이벤트를 볼 수 있다.
- [x] 메인 페이지에서 카드를 클릭하면 상세 이벤트 보기 페이지로 이동할 수 있다.
- [x] 사용자는 참여 버튼을 누르면 이벤트에 참여할 수 있다.
- [x] 사용자는 참여 완료 시 리뷰하기 버튼이 생긴다.
- [x] 사용자는 리뷰하기 버튼을 클릭 시 리뷰를 할 수 있다.
- [x] 사용자는 좋아요를 누를 시 이벤트를 좋아요로 추가할 수 있다.
- [x] 사용자는 좋아요 취소를 누를 시 이벤트의  좋아요를 취소할 수 있다.
- [x] 사용자는 즐겨찾기를 누를 시 이벤트의 가게를 즐겨찾기 할 수 있다.
- [x] 사용자는 즐겨찾기 취소를 누를 시 이벤트의 가게 즐겨찾기를 취소할 수 있다.
- [x] 사용자는 주소를 수정할 수 있다.
- [x] 사용자는 리뷰 리스트를 볼 수 있다.
- [x] 사용자는 리뷰를 클릭하면 리뷰 전체 내역을 볼 수 있다.
- [x] 리뷰가 없는 이벤트 상세 조회 페이지에는 기본적으로 리뷰를 장려하는 카드가 보인다.

JengYoung committed:
fix: 스토리북 빌드 시 오류 수정
